### ISSUE TITLE
fix: 🤔 remove `tabindex`on non-closable sd-notification

### ISSUE THREAD

**Issue by paulovareiro29** [solid-design-system/solid#2447]
## Current behavior
The `sd-notification` has `tabindex=0` even when it is not interactive (not closable)

## Expected behavior
When the notification is not interactive (not closable), it shouldn't be tabbable.
If there is an interactive element inside the notification, the element should be tabbable, which is just following the normal browser behaviour.

## Technical Information
- [ ] Remove `tabindex=0` when notification is not closable.

## DoR
- [ ] Item has business value
- [ ] Item has been estimated by the team
- [ ] Item is clear and well-defined
- [ ] Item dependencies have been identified
